src/components/MultiFilterButton/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { Link } from 'react-router-dom'
+import { Popover } from 'antd'
+import { useTranslation } from 'react-i18next'
+import { ReactComponent as FilterIcon } from '../../assets/filter_icon.svg'
+import { ReactComponent as SelectedIcon } from '../../assets/selected-icon.svg'
+import { ReactComponent as NotSelectedIcon } from '../../assets/not-selected-icon.svg'
+import { ReactComponent as PartialSelectedIcon } from '../../assets/partial-selected-icon.svg'
+import { useSearchParams } from '../../hooks'
+import styles from './styles.module.scss'
+
+export function MultiFilterButton({
+  filteredList,
+  isMobile,
+  filterName,
+}: {
+  filterName: string
+  filteredList: { key: string; value: string; to: string; title: string | JSX.Element }[]
+  isMobile?: boolean
+}) {
+  const { t } = useTranslation()
+  const params = useSearchParams(filterName)
+  const filter = params[filterName]
+  let types = filteredList.map(f => f.value)
+  if (filter !== undefined) {
```

Comment:
It can be simplified as
```ts
const types = params[filterName]?.split(',').filter(t => !t)
```

Reply:
`filterName` should be `filterNames` if it's designed to be multiple filters.

Reply:
For me, the `filters` from the URL are more like `filteredList` because they are effective/working `filters` while `filteredList` in the `component params` is a list of filters.

Reply:
> `filterName` should be `filterNames` if it's designed to be multiple filters.

I think there would be only one filter for each filter button, so I think "s" is not necessary here.

Reply:
There are three values for the `filterName`.
1. `undefined`, which means to entry the page without any filter parameters, for "all selected".
2. Empty string `''`, like `tags=` which means "not selected any value".
3. Partial selected.

Reply:
> There are three values for the `filterName`.
> 
> 1. `undefined`, which means to entry the page without any filter parameters, for "all selected".
> 2. Empty string `''`, like `tags=` which means "not selected any value".
> 3. Partial selected.

```ts
const types = params[filterName]?.split(',').filter(t => !t)
```

works well for these 3 cases

when `params[filterName]` is undefined, `types` is undefined, means `all selected`
when `params[filterName]` is an empty string, `types` is an empty array, means `not selected any value`
when `params[filterName]` is a string, `types` is expected to be an array with items, means `some selected`

Reply:
Any update on this conversation?

Reply:
> > > `filterName` should be `filterNames` if it's designed to be multiple filters.
> > 
> > 
> > I think there would be only one filter for each filter button, so I think "s" is not necessary here.
> 
> The logic is as follows
> 
> ```ts
> const filter = params[filterName]
> const types = filter.split(',').filter(t => t !== '')
> ```
> 
> equivalent to
> 
> ```ts
> const types = params[filterName].split(',').filter(t => t !== '')
> ```
> 
> So `params[filterName]` is expected to be a string like `A,B,C`, which includes multiple `values`. That's the reason I think it should be `filterNames`

The `filterName` means the name of the filter, and `A,B,C` should be the values of this filter, not the name.

Reply:
> > There are three values for the `filterName`.
> > 
> > 1. `undefined`, which means to entry the page without any filter parameters, for "all selected".
> > 2. Empty string `''`, like `tags=` which means "not selected any value".
> > 3. Partial selected.
> 
> ```ts
> const types = params[filterName]?.split(',').filter(t => !t)
> ```
> 
> works well for these 3 cases
> 
> when `params[filterName]` is undefined, `types` is undefined, means `all selected` when `params[filterName]` is an empty string, `types` is an empty array, means `not selected any value` when `params[filterName]` is a string, `types` is expected to be an array with items, means `some selected`

The result of `!t` is `true` when it is an empty string, but the empty string should filtered to the result

Reply:
My typo, `.filter(t => !t)` should be `.filter(t => !!t)` to return truty values